contrib/lua/premake5.lua

```diff
@@ -21,8 +21,8 @@ project "lua-lib"
 		"etc/*.c"
 	}
 
-	configuration "linux or bsd or hurd or aix"
+	filter "system:linux or system:bsd or system:hurd or system:aix"
```

Comment:
Sorry, I just realised what you meant by the sequence of `or`'s. `not x or y or z`, I see `not` doesn't gel with `or` very well. The feeling I get from my comment above still stands though, that `not` to the left of the scope doesn't make sense to me.

Working `and` into those expressions doesn't seem too unreasonable, but that inevitably leads to parentheses... that starts to seem like too much, but I have always found it a bit strange that `or` is stated within the string, and `and` is expressed by comma separation outside the string.

I think there's probably way to much water under the bridge to change anything now,  but I do wonder if a more regular or intuitive expression might be possible.